src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -40,18 +40,29 @@ class DoctrineChoiceLoader implements ChoiceLoaderInterface
      * passed which optimizes the object loading for one of the Doctrine
      * mapper implementations.
      *
-     * @param ObjectManager              $manager      The object manager
-     * @param string                     $class        The class name of the loaded objects
-     * @param IdReader                   $idReader     The reader for the object IDs
-     * @param EntityLoaderInterface|null $objectLoader The objects loader
+     * @param ObjectManager              $manager           The object manager
+     * @param string                     $class             The class name of the loaded objects
+     * @param IdReader                   $idReader          The reader for the object IDs
+     * @param EntityLoaderInterface|null $objectLoader      The objects loader
+     * @param bool                       $checkOptimization For BC, to be removed in 5.0
      */
-    public function __construct(ObjectManager $manager, string $class, IdReader $idReader = null, EntityLoaderInterface $objectLoader = null)
+    public function __construct(ObjectManager $manager, string $class, IdReader $idReader = null, EntityLoaderInterface $objectLoader = null, bool $checkOptimization = true)
     {
         $classMetadata = $manager->getClassMetadata($class);
 
+        if ($checkOptimization && null === $idReader) {
+            $idReader = new IdReader($manager, $classMetadata);
+
+            if ($idReader->isSingleId()) {
+                @trigger_error(sprintf('Not explicitly passing an instance of "%s" when it can optimize single id entity "%s" has been deprecated in 4.3 and will apply any optimization in 5.0.', IdReader::class, $class), E_USER_DEPRECATED);
```

Comment:
I don't understand the `and will apply any optimization in 5.0` part. It will apply something or it won't?

Reply:
I added the missing "not" :), thanks!